src/libstore/build.cc

```diff
@@ -2045,6 +2047,11 @@ void DerivationGoal::startBuilder()
                 }
     }
 
+#if DARWIN_PREBUILD
+    if (settings.preBuildHook == "")
+      settings.preBuildHook = settings.nixLibexecDir + "/nix/resolve-system-dependencies.pl";
```

Comment:
This (plus associated macros) should be in the `Settings::Settings()` constructor in globals.cc